tools/src/main/java/org/apache/kafka/tools/VerifiableConsumer.java

```diff
@@ -664,7 +664,11 @@ public static VerifiableConsumer createFromArgs(ArgumentParser parser, String[]
 
         consumerProps.put(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, useAutoCommit);
         consumerProps.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, res.getString("resetPolicy"));
-        consumerProps.put(ConsumerConfig.SESSION_TIMEOUT_MS_CONFIG, Integer.toString(res.getInt("sessionTimeout")));
+
+        // session.timeout.ms cannot be set when group.protocol=CONSUMER
+        if (!groupProtocol.equalsIgnoreCase(GroupProtocol.CONSUMER.name)) {
```

Comment:
If we set `--session-timeout` for consumer protocol, can we add warning for this config?

Reply:
Thanks for the suggestion! However, I’m uncertain whether this change is necessary. When group.protocol=CONSUMER is used, setting a session timeout in the consumer prevents it from starting successfully. This makes me feel the change might be redundant. That said, I’d appreciate hearing others’ thoughts on this. Thanks!

Reply:
maybe we should call `consumerProps.put(ConsumerConfig.SESSION_TIMEOUT_MS_CONFIG, xx)` only if users do set the `--session-timeout` - that means we should remove the default value of  `--session-timeout`

Reply:
Sounds good, btw we already set `--session-timeout` 30sec by default in https://github.com/apache/kafka/blob/440e0b8801bca5a648a2cc2a7e98f25c44b0810d/tests/kafkatest/services/verifiable_consumer.py#L234

Reply:
The default value of `session.timeout.ms` is 45 seconds, so I'm unsure if it's necessary to set it to 30 seconds in the end-to-end tests. Could you please remove this setting from the e2e configuration? Additionally, e2e should not use `--session-timeout` when running AsyncConsumer.

Reply:
Thanks, I already addressed the comment in the latest commit.